http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/DataprocInterpreter.scala

```diff
@@ -424,12 +425,10 @@ class DataprocInterpreter[F[_]: Timer: Parallel: ContextShift](
           F.unit
         } else {
           for {
-            projectNumberOptIO <- F.liftIO(
-              IO.fromFuture(IO(googleProjectDAO.getProjectNumber(googleProject.value)))
-            )
-            projectNumber <- F.liftIO(
-              IO.fromEither(projectNumberOptIO.toRight(ClusterIamSetupException(imageProject)))
-            )
+            projectNumberOpt <- googleResourceService.getProjectNumber(googleProject)
+
+            //TODO: This can also be from a 404?
+            projectNumber <- F.fromEither(projectNumberOpt.toRight(ClusterIamSetupException(imageProject)))
```

Comment:
So, I'm not really sure in what circumstances the exception here occurs.

I wanted to see if anyone knew what it was for (@rtitle ?) before I touchedd it, but I added a TODO there with my thoughts. I'm thinking this should be a project not found exception. As in, the library returns `Option[Long]` now, where it is none if the project passed is not found. Any idea if the exception here is still relevant?

Reply:
my guess is this was just for dealing with the case when project doesn't exist...does your `googleResourceService.getProjectNumber` return `None` when project doesn't exist currently?

Reply:
yeah, I'm thinking I should change the exception here

Reply:
@rtitle just wonder if you have more context around this code...